lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -95,17 +101,48 @@ export class AuthorizationCodeModule extends AuthModule {
         if (!codeResponse || !codeResponse.code) {
             throw ClientAuthError.createAuthCodeNullOrEmptyError();
         }
+
         let encodedTokenRequest;
+        let tokenRequest: TokenExchangeParameters;
         if (!request) {
             encodedTokenRequest = this.cacheStorage.getItem(TemporaryCacheKeys.REQUEST_PARAMS);
+            try {
+                tokenRequest = JSON.parse(this.cryptoObj.base64Decode(encodedTokenRequest)) as TokenExchangeParameters;
+                this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_PARAMS);
+            } catch (err) {
+                throw ClientAuthError.createTokenRequestCacheError(err);
+            }
+        } else {
+            tokenRequest = request;
```

Comment:
Alternative, which doesn't use mutation and a bit easier to read (IMO):

```js
const getCachedRequest = () => {
  try {
    const encodedTokenRequest = this.cacheStorage.getItem(TemporaryCacheKeys.REQUEST_PARAMS);
    const parsedRequest = JSON.parse(this.cryptoObj.base64Decode(encodedTokenRequest)) as TokenExchangeParameters;
    return parseRequest;
  } catch (err) {
     throw ClientAuthError.createTokenRequestCacheError(err);
  }
};
const tokenRequest = request || getCachedRequest();
```

Reply:
Done - made a getCachedRequest helper function